oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/authentication/logout/OidcBackChannelLogoutFilter.java

```diff
@@ -0,0 +1,205 @@
+
+	private RequestMatcher requestMatcher = new AntPathRequestMatcher("/logout/connect/back-channel/{registrationId}",
+			"POST");
+
+	private OidcProviderSessionRegistry providerSessionRegistry = new InMemoryOidcProviderSessionRegistry();
+
+	private LogoutHandler logoutHandler = new BackchannelLogoutHandler();
+
+	/**
+	 * Construct an {@link OidcBackChannelLogoutFilter}
+	 * @param clients the {@link ClientRegistrationRepository} for deriving Logout Token
+	 * validation
+	 * @param logoutTokenDecoderFactory the {@link JwtDecoderFactory} for constructing
+	 * Logout Token validators
+	 */
+	public OidcBackChannelLogoutFilter(ClientRegistrationRepository clients,
+			AuthenticationManager authenticationManager) {
+		this.clientRegistrationRepository = clients;
+		this.authenticationManager = authenticationManager;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+			throws ServletException, IOException {
+		RequestMatcher.MatchResult result = this.requestMatcher.matcher(request);
+		if (!result.isMatch()) {
+			chain.doFilter(request, response);
+			return;
+		}
+		String registrationId = result.getVariables().get("registrationId");
+		ClientRegistration registration = this.clientRegistrationRepository.findByRegistrationId(registrationId);
+		if (registration == null) {
+			this.logger.debug("Did not process OIDC Backchannel Logout since no ClientRegistration was found");
+			chain.doFilter(request, response);
+			return;
+		}
+		String logoutToken = request.getParameter("logout_token");
+		if (logoutToken == null) {
+			String error = "Failed to process OIDC Backchannel Logout since no logout token was found";
+			this.logger.debug(error);
+			String message = String.format(ERROR_MESSAGE, "invalid_request", error);
+			response.sendError(400, message);
+			return;
+		}
+		OidcLogoutToken token;
+		try {
+			token = authenticate(logoutToken, registration);
+		}
+		catch (AuthenticationException ex) {
+			this.logger.debug("Failed to process OIDC Backchannel Logout", ex);
+			String message = String.format(ERROR_MESSAGE, "invalid_request", ex.getMessage());
+			response.sendError(400, message);
+			return;
+		}
+		int sessionCount = 0;
+		int loggedOutCount = 0;
+		List<String> messages = new ArrayList<>();
+		Iterable<OidcProviderSessionRegistrationDetails> sessions = this.providerSessionRegistry.deregister(token);
+		for (OidcProviderSessionRegistrationDetails session : sessions) {
+			BackchannelLogoutAuthentication authentication = new BackchannelLogoutAuthentication(
+					session.getClientSessionId(), session.getCsrfToken());
+			try {
+				if (this.logger.isTraceEnabled()) {
+					String message = "Logging out session #%d from result set for issuer [%s]";
+					this.logger.trace(String.format(message, sessionCount, token.getIssuer()));
+				}
+				this.logoutHandler.logout(request, response, authentication);
```

Comment:
I'm not following your logic. How is it that the thing that is handling the logout is not a logout handler? By the same logic (if I understand you correctly), `LogoutFilter` should not use `LogoutHandler`s.

The purpose of an authentication manager is to authenticate materials. The purpose of a logout handler is to perform logouts.

Other filters have handler components based on the result of the authentication. For example, login filters authenticate have authentication success and failure handlers. This does not violate the principle that these filters are controllers.

Finally, I don't think that the authentication manager can perform the logout since it requires the `HttpServletRequest` to acquire the target endpoint.